x-conduit/src/X/Control/Monad/Trans/Resource.hs

```diff
@@ -0,0 +1,23 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module X.Control.Monad.Trans.Resource where
+
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.Resource (MonadResource, liftResourceT)
+import Control.Monad.Trans.Either (EitherT)
+
+-- | This instance is useful when adding errors to the monad used for some conduit processing
+-- when resources, like files, are involved. For example:
+--
+-- foo :: MonadResource m => ConduitM () Void (EitherT Text m) [ByteString]
+-- foo = CB.sourceFile "src/Main.hs"
+--  =$= CB.lines
+--  -- filter some lines
+--  =$= CL.map parse
+--  -- collect errors in EitherT Text m
+--  =$= CL.mapM hoistEither
+--  =$= CL.consume
+--
+-- parse :: ByteString -> Either Text ByteString
+-- parse _ = Left ":("
+instance MonadResource m => MonadResource (EitherT e m) where
```

Comment:
Ed Kmett explains the [situation](https://www.reddit.com/r/haskell/comments/3ded39/why_cant_we_have_an_eithert_in_transformers/ct4mnk1), I wasn't aware of this, thought others might be interested.